src/gwrefpy/model.py

```diff
@@ -216,71 +216,137 @@ def _fit(
 
-    def _best_fit(self, ref_well=None, obs_well=None):
+    def _best_fit(
+        self,
+        obs_well: str | Well,
+        ref_wells: list[str | Well] | None = None,
+        method: Literal["linearregression"] = "linearregression",
+        **kwargs,
+    ) -> FitResultData:
         """
         The internal method to find the best fit.
 
         Parameters
         ----------
-        ref_well : Well or list of Well or None, optional
-            The reference well to use for fitting (default is None).
         obs_well : Well or list of Well or None, optional
-            The observation well to use for fitting (default is None).
+            The observation well to use for fitting.
+        ref_wells : Well or list of Well or None, optional
+            The reference wells to test. If None, all reference wells in the
+            model will be used (default is None).
+        method : Literal["linearregression"]
+            Method with which to perform regression. Currently only supports
+            linear regression.
+        **kwargs
+            Keyword arguments to pass to the fitting method. For example, you can use
+            `offset`, `p`, `tmin`, and `tmax` to control
 
         Returns
         -------
-        None
-            This method modifies the model in place.
+        FitResultData
+            Returns the best fit for the given arguments.
         """
-        # Placeholder for internal best fit logic
-        logger.debug(f"Internal best fit logic for model '{self.name}'.")
-        # best_fit = BestFitResults(ref_well, obs_well)
-        best_fit = "h"
-        self.fits.append(best_fit)
-        logger.info(f"Best fit completed for model '{self.name}'.")
-
-    def get_fits(self, well: Well):
+        if isinstance(ref_wells, list) and len(ref_wells) < 1:
+            logger.error("ref_wells list cannot be empty.")
+            raise ValueError("ref_wells list cannot be empty.")
+
+        if isinstance(obs_well, str):
+            target_obs_well = self.get_wells(obs_well)
+            if isinstance(target_obs_well, list):
+                raise ValueError(
+                    "obs_well parameter must resolve to a single well, not a list."
+                )
+        elif isinstance(obs_well, Well):
+            target_obs_well = obs_well
+        if ref_wells is None:
+            target_ref_wells = [well for well in self.wells if well.is_reference]
+            if len(target_ref_wells) < 1:
+                logger.error("No reference wells available in the model.")
+                raise ValueError("No reference wells available in the model.")
+        else:
+            target_ref_wells: list[Well] = []
+            for rw in ref_wells:
+                if isinstance(rw, str):
+                    target_ref_wells.append(self.get_wells(rw))  # type: ignore
+                elif isinstance(rw, Well):
+                    target_ref_wells.append(rw)
```

Comment:
```suggestion
                    target_ref_wells.append(rw)
                else:
                    logger.error(f"Unsupported type for {rw}. Supported types are Well or str")
                    raise TypeError(f"Unsupported type for {rw}. Supported types are Well or str")

```
Or us this check implemented elsewhere?